### PR TITLE
Convert issue templates from GHFM to YAML.

### PR DESCRIPTION
**Pull request by RokeJulianLockhart**
Fixes https://github.com/BOINC/boinc/issues/5940#issuecomment-2531429972.

#### **Description of the Change**

1. ~~Makes the headings semantic.~~
2. Makes the placeholders invisible outside of the editor by default, so that new users cannot be confused about what from the template needs to be retained.

#### **Alternate Designs**

The current version is the sole alternative. It was used because it controlled the appearance of the heading text. However, `<h4>` on GitHub's is `1em` too, and `<b>` can still be applied (in case it's ever removed from the CSS) without it applying twice until then.

#### **Release Notes**

No need.

### CODE REVIEW

**Comment by AenBleidd** [comment]
Please make it in YAML as I suggested. Thank you in advance.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/BOINC/boinc/pull/5944#issuecomment-2531928207

@AenBleidd, I'm not sure how to test it on my repository, though. I can't seem to force GitHub to choose the issue template from the patch branch, instead of `origin/HEAD`. I can submit my best estimate in YAML format, but it'll be untested.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/BOINC/boinc/pull/5944/commits/8c0e1cb2afd84991b3f5ea31b6e95574384be08d

1. I've duplicated [`microsoft/vcpkg/blob/a1a1cbc975abf909a6c8985a6a2b8fe20bbd9bd6/.github/ISSUE_TEMPLATE/04-request-a-feature-or-improvement.yml`](https://github.com/microsoft/vcpkg/blob/a1a1cbc975abf909a6c8985a6a2b8fe20bbd9bd6/.github/ISSUE_TEMPLATE/04-request-a-feature-or-improvement.yml#L1-L29:~:text=name:%20Request%20a%20feature%20or,required:%20false);
1. I've removed the invalid key (`title`) per [`orgs/community/discussions/4627#discussioncomment-1021637`](https://github.com/orgs/community/discussions/4627#discussioncomment-1021637).
1. I've removed the `label` key, because current issues aren't labelled by default.

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Feature request
+description: Suggest an idea for this project.
+body:
+  - type: textarea
```
Comment (AenBleidd) [inline .github/ISSUE_TEMPLATE/feature_request.yml:4]: Should be marked as `required`

**Review by AenBleidd — CHANGES_REQUESTED** [review]
So far looks good.
Could you please convert the other two templates as well?
Thank you in advance.

**Comment by RokeJulianLockhart** [comment]
> Could you please convert the other two templates as well?

@AenBleidd, I don't mind doing that - might as well. I'll probably get to it tomorrow.

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 1 above)
Comment (RokeJulianLockhart) [reply .github/ISSUE_TEMPLATE/feature_request.yml:4]: > https://github.com/BOINC/boinc/pull/5944#discussion_r1878880787

@AenBleidd, I've committed [`commits/15752cec4889f770caf2c80ab1155b7f6e4865f3#diff-c6b098646bf32c644234bec14c2675ea2a3d9c2dc2df450a25aa0e7ff02323cdR9-R21`](https://github.com/BOINC/boinc/pull/5944/commits/15752cec4889f770caf2c80ab1155b7f6e4865f3#diff-c6b098646bf32c644234bec14c2675ea2a3d9c2dc2df450a25aa0e7ff02323cdR9-R21:~:text=required:%20false-,required:%20true,%2D%20type:%20textarea,-%2D%20type:%20textarea&text=%2D%20type:%20textarea-,%2D%20type:%20textarea,attributes:,-attributes:&text=attributes:-,attributes:,Describe%20the%20solution%20you'd%20like,-label:%20Describe%20the&text=solution%20you'd%20like-,label:%20Describe%20the%20solution%20you'd,you%20want%20to%20happen.%20%2D%2D%3E%22,-placeholder:%20%22%3C!%2D%2D%20A&text=to%20happen.%20%2D%2D%3E%22-,placeholder:%20%22%3C!%2D%2D%20A%20clear%20and,validations:,-validations:&text=validations:-,validations:,required:%20false,-required:%20true&text=required:%20false-,required:%20true,%2D%20type:%20textarea,-%2D%20type:%20textarea&text=%2D%20type:%20textarea-,%2D%20type:%20textarea,attributes:,-attributes:&text=attributes:-,attributes:,label:%20Additional%20context,-label:%20Additional%20context&text=label:%20Additional%20context-,label:%20Additional%20context,or%20features%20you've%20considered.%20%2D%2D%3E%22,-placeholder:%20%22%3C!%2D%2D%20A&text=you've%20considered.%20%2D%2D%3E%22-,placeholder:%20%22%3C!%2D%2D%20A%20clear%20and,validations:,-validations:&text=validations:-,validations:,required:%20false,-required:%20true&text=required:%20false-,required:%20true,-Footer).

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      label: Describe the problem
+      placeholder: "<!-- A clear and concise description of what the problem is. Ex. I'm always frustrated when [...] -->"
+    validations:
+      required: true
+  - type: textarea
+    attributes:
+      label: Describe the solution you'd like
+      placeholder: "<!-- A clear and concise description of what you want to happen. -->"
+    validations:
+      required: true
```
Comment (AenBleidd) [inline .github/ISSUE_TEMPLATE/feature_request.yml:15]: This should be left optional

File: .github/ISSUE_TEMPLATE/feature_request.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  - type: textarea
+    attributes:
+      label: Additional context
+      placeholder: "<!-- A clear and concise description of any alternative solutions or features you've considered. -->"
+    validations:
+      required: true
```
Comment (AenBleidd) [inline .github/ISSUE_TEMPLATE/feature_request.yml:21]: This also should be left optional

**Review by AenBleidd — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/feature_request.yml (hunk 1 above)
Comment (AenBleidd) [reply .github/ISSUE_TEMPLATE/feature_request.yml:4]: I see your commits. No need to link them to your replies.

**Comment by RokeJulianLockhart** [comment]
> https://github.com/BOINC/boinc/pull/5944/commits/a994bb4097e2d0612caff51ad3d339a911446bad

I didn't commentify the two multiline strings, because they're templates (which to be utilized), rather than placeholder text (which is to be ignored or removed).

**Comment by RokeJulianLockhart** [comment]
> https://github.com/BOINC/boinc/pull/5944#event-15607897845

@AenBleidd, I wonder whether the placeholder text can be converted into body text beneath the headings, instead. As it is, they seem to render at [`RokeJulianLockhart/boinc/blob/f63b77589fe36edd359db88a414c5738d1b46849/.github/ISSUE_TEMPLATE/other-issues.yml`](https://github.com/RokeJulianLockhart/boinc/blob/f63b77589fe36edd359db88a414c5738d1b46849/.github/ISSUE_TEMPLATE/other-issues.yml#:~:text=any%20other%20issues.-,Describe%20the%20bug,-*) as escaped source markup, so perhaps commenting them is unnecessary.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
+name: Bug report
```
Comment (AenBleidd) [inline .github/ISSUE_TEMPLATE/bug_report.yml:1]: image

**Review by AenBleidd — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+      required: false
+  - type: textarea
+    attributes:
+      label: Screenshots
+      placeholder: "<!-- If applicable, add screenshots to help explain your problem. -->"
+    validations:
+      required: false
+  - type: textarea
+    attributes:
+      label: Steps to reproduce
```
Comment (AenBleidd) [inline .github/ISSUE_TEMPLATE/bug_report.yml:32]: Should be `System information` instead (and it should resolve the issue on the screenshot above)

Comment (RokeJulianLockhart) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Apologies. That wasn't deliberate.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 4 above)
Comment (RokeJulianLockhart) [reply .github/ISSUE_TEMPLATE/bug_report.yml:1]: It doesn't show that anymore.

**Comment by RokeJulianLockhart** [comment]
image
image
image

**Review by AenBleidd — APPROVED** [review]

**Comment by AenBleidd** [comment]
@RokeJulianLockhart, thank you for the contribution!

**Comment by RokeJulianLockhart** [comment]
@AenBleidd, thank you for your assistance.

**Pull request merged**